package.json

```diff
@@ -2,47 +2,53 @@
   "name": "iter-tools",
   "version": "4.0.0",
   "description": "The iterator toolbox",
-  "main": "lib/index.js",
-  "module": "lib/index.mjs",
-  "typings": "lib/index.d.ts",
+  "main": "src/index.js",
+  "module": "src/index.mjs",
+  "typings": "src/index.d.ts",
   "scripts": {
     "test": "npm run build && mocha tests/**/*.js",
-    "benchmarks": "node benchmarks",
+    "benchmarks": "npm run build && node benchmarks",
     "watch": "npm run test -- -w",
     "lint": "eslint --fix --ext .mjs ./src ./tests ./benchmarks",
-    "clean": "rm -rf lib/*.* && rm -rf es5/*.* && rm -rf package/*",
-    "build": "npm run clean && cross-env BABEL_ENV=es6 babel -x \".mjs\" src -d lib && renamer --regex --find '\\.js$' --replace '.mjs' 'lib/**' && cross-env BABEL_ENV=es5 babel -x \".mjs\" src -d es5 && cross-env BABEL_ENV=cjs babel -x \".mjs\" src -d lib && cp src/*.d.ts lib",
-    "dist": "npm run build && cp -R package.json lib/* es5 LICENSE README.md package/ && cd package && json -I -f package.json -e \"this.main='index.js';this.typings='index.d.ts';this.module='index.mjs'\" && npm publish",
+    "clean": "rimraf dist",
+    "build": "npm run clean && npm run build:cjs && npm run build:es5 && npm run build:es5-cjs && npm run build:es2015 && npm run build:es2015-cjs && cp src/*.d.ts dist",
+    "build:cjs": "cross-env BABEL_ENV=cjs babel -x \".mjs\" src -d dist && copyfiles -f src/*.mjs dist/",
```

Comment:
I must say again that you almost certainly should not use the untranspiled files in the root of the package.

No transpilers touch code in node_modules (which iter-tools will be). However, static analysis, tree shaking, minificiation, etc, all need to be done on these files, and will fail unless the user's toolchain happens to be configured for nonstandard language features.

Finally when the untranspiled code makes it to the browser, in the event that the browser can't handle the syntax it will fail parsing /the entire bundle/. No code runs at all, and users start asking why they have just a blank white screen...

IE11 will also junk the bundle if someone packages iter-tools without reading the documentation on using the /es5 folder. The docs would have to say that you will almost certainly want to use es5 unless you definitely know what you're doing. I doubt that most people who in theory need to support IE11 actually bother testing it very often, so it could be a while before they notice that their whole app is gone.

Embrace the principle of least surprise!

Reply:
Would you put the es5 version in the root?

Reply:
By my own logic, yes. I don't love it because the files are so large and gross, but I don't think the other options are better. At least it means that everyone can write their code the way they want without having to worry about IE11, and I don't hope to spend any more of my life worrying about IE11.